utilities/pipelines/PSRuleValidation/Set-PSRuleOutput.ps1

```diff
@@ -0,0 +1,156 @@
+﻿<#
+.SYNOPSIS
+Parse an input csv file containing the output of the PSRule pre-flight checks and generate formatted markdown file out of it.
+
+.DESCRIPTION
+Parse input csv file containing the output of the PSRule pre-flight checks and generate formatted markdown file out of it.
+
+.PARAMETER inputFilePath
+Mandatory. The path to the output file created by PSRule in csv format.
+
+.PARAMETER outputFilePath
+Optional. The path to the formatted .md file to be created.
+
+.PARAMETER skipPassedRulesReport
+Optional. Whether to add the detail of passed PSRule to the output markdown file or to limit the list to the failed ones.
+
+.EXAMPLE
+Set-PSRuleOutput -inputFilePath 'C:/PSRule-output.csv'
+
+Generate a markdown file 'output.md' in the current folder, out of the 'C:/PSRule-output.csv' input, listing all passed and failed rules.
+
+.EXAMPLE
+Set-PSRuleOutput -inputFilePath 'C:/PSRule-output.csv' -outputFilePath 'C:/PSRule-output.md' -skipPassedRulesReport
+
+Generate a markdown file 'C:/PSRule-output.md', out of the 'C:/PSRule-output.csv' input, listing only the failed rules.
+#>
+function Set-PSRuleOutput {
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory)]
+        [String] $inputFilePath,
+
+        [Parameter(Mandatory = $false)]
+        [string] $outputFilePath = './output.md',
+
+        [Parameter(Mandatory = $false)]
+        [switch] $skipPassedRulesReport
+    )
+
+    ###########################################
+    # Import CSV output and filter by results #
+    ###########################################
+
+    $results = Import-Csv -Path $inputFilePath
+
+    $passedRules = @()
+    $failedRules = @()
+
+    $passedRules += $results | Where-Object { $_.Outcome -EQ 'Pass' }
+    $failedRules += $results | Where-Object { $_.Outcome -EQ 'Fail' }
+
+    ######################
+    # Set output content #
+    ######################
+
+    # Header
+    $header = [System.Collections.ArrayList]@(
+        '# PSRule pre-flight validation summary ',
+        ''
+    )
+    Out-File -FilePath $outputFilePath -NoClobber -InputObject $header
+
+    if ($failedRules.Count -eq 0) {
+        # No failure content
+        $noFailuresContent = ('## :rocket: All {0} rules passed, YAY! :rocket:' -f $results.Count)
```

Comment:
I usually like to emphasize the 'variable' value. But I'll leave it up to you to agree, or reject the suggestion as it super subjective ;)

```suggestion
        $noFailuresContent = ('## :rocket: All [{0}] rules passed, YAY! :rocket:' -f $results.Count)
```